### PR TITLE
fix(swc/cli): update plugin template for pinned dev

### PR DESCRIPTION
**Pull request by kwonoj**
**Description:**

Removes pinned deps from template as well for plugin generation.

I think this PR allows us to reach initial completion of https://github.com/swc-project/swc/issues/3167 (🎉 )

Once this PR goes in, I'd like to do

1. mark #3167 as complete
2. write a feedback collection post at https://github.com/swc-project/swc/discussions

3. probably prepare some blog post / twitter to ask users to try this one

Thoughts @kdy1 ? 3 may need some thoughts on how to approach those.

### CODE REVIEW

**Comment by kdy1** [comment]
I think your plan is good.
For some reason auto-merge button is disappeared from github, so I'll merge it manualy once CI passes.

**Review by kdy1 — APPROVED** [review]

**Comment by kwonoj** [comment]
Sounds good. Let me do 1/2, and discuss 3 separately. I can't write blog post myself anyway 😅

**Pull request merged**
